amf-shapes/shared/src/main/scala/amf/shapes/internal/spec/jsonldschema/parser/JsonLDPropertyParser.scala

```diff
@@ -116,10 +116,16 @@ case class JsonLDPropertyParser(
       ctx.eh.violation(ContainerCheckErrorList, range, ContainerCheckErrorList.message)
 
   private def computeTerm(path: JsonPath): String = {
-    propertyName(path)
-      .flatMap(semanticContext.getTermFor)
-      .map(semanticContext.expand)
-      .getOrElse(semanticContext.computeBase(path.toString))
+    propertyName(path) match {
+      case Some(name) =>
+        semanticContext
+          .getTermFor(name)
+          .map(semanticContext.expand)
+          .getOrElse(semanticContext.computeBase(name))
+
+      // should be unreachable
+      case _ => throw new IllegalStateException(s"Property ${path.toString} does not have name")
```

Comment:
[Out of Scope] - This is shady, we should have a safe-compiler way of doing this check (I don't know how at the moment)